test/object-store/collection_fixtures.hpp

```diff
@@ -34,25 +34,13 @@
 
 namespace realm::collection_fixtures {
 
-struct less {
-    template <class T, class U>
-    auto operator()(T&& a, U&& b) const noexcept
-    {
-        return Mixed(a).compare(Mixed(b)) < 0;
-    }
-};
-struct greater {
-    template <class T, class U>
-    auto operator()(T&& a, U&& b) const noexcept
-    {
-        return Mixed(a).compare(Mixed(b)) > 0;
-    }
-};
+template <typename T>
+constexpr bool always_false = false;
 
 template <typename T>
 inline T get(Mixed)
 {
-    abort();
```

Comment:
The main goal of these changes is to eliminate these aborts and replace them with compile-time failures. As a bonus it ends up less verbose and actually compilers faster due to eliminating unreachable template instantiations.